### DISCUSSION TITLE
Class Pelago\Emogrifier does not exist

### DISCUSSION THREAD

**Discussion by bkraut** [scandipwa/scandipwa#3226] [Install & Upgrade] [answered]
Hi,
I installed ScandiPWA 3 times. Everithing goes smooth until I try to run bin/magento setup:di:compile. Then I get this error:

`  Class Pelago\Emogrifier does not exist
  Class ScandiPWA\CmsGraphQl\Model\Template\Filter\Interceptor generation error: The requested class did not generate properly, becaus
  e the 'generated' directory permission is read-only. If --- after running the 'bin/magento setup:di:compile' CLI command when the 'g
  enerated' directory permission is set to write --- the requested class did not generate properly, then you must add the generated cl
  ass object to the signature of the related construct method, only.
`
I ran composer update and everithing is on the latest version.
What could be a reason?

Best,
Bojan

**Comment by zans-laksa** [comment, marked as answer]
Please downgrade the magento to 2.4.2.
This will be fixed when magento 2.4.3 support will be released.
For more details please see https://github.com/scandipwa/cms-graphql/pull/15
